### ISSUE TITLE
how to add another wake word to "stop" 

### ISSUE THREAD

**Issue by hemsinghBanoth** [ShawnHymel/tflite-speech-recognition#2]
hello,

I tried this code its working fine .. only i have doubt is how to add an other wake word to the code below.

site address:  
https://github.com/ShawnHymel/tflite-speech-recognition/blob/master/02-speech-commands-mfcc-classifier.ipynb

1. 
# Settings
feature_sets_path = 'C:\\PATH\\TO\\feature_sets_directory'
feature_sets_filename = 'all_targets_mfcc_sets.npz'
model_filename = 'wake_word_stop_model.h5'
wake_word = 'stop'
  is it 
wake_word = 'stop', "yes"   and so on... or  other way .

2. # Convert ground truth arrays to one wake word (1) and 'other' (0)
wake_word_index = all_targets.index(wake_word)
y_train = np.equal(y_train, wake_word_index).astype('float64')
y_val = np.equal(y_val, wake_word_index).astype('float64')
y_test = np.equal(y_test, wake_word_index).astype('float64')

what about second (2nd) wake word ?  how to proceed ...?

please let me know .

with regards.
Hem

**Comment by ShawnHymel** [comment]
The model only supports 1 wake word right now. I'm working on having it support multiple words.

**Comment by nishant2019** [comment]
I am trying to use the code for detection of noise and silence and have created sound samples for 10 seconds each. However all the samples gets dropped due **mfccs.shape** being **Dropped: 1 (16, 196)**. What can be the error. ?

**Comment by mopbop** [comment]
These instructions will help you turn this model into a multiclass speech command detector

1. In the "01-speech-commands-mfcc-extraction.ipynb" file, train only the words you want to 


2. In the "02-speech-commands-mfcc-classifier.ipynb" file, look for the step "Convert ground truth arrays to one wake word (1) and 'other' (0)". 
Replace the code below.
Change the num_classes to the number of words you trained the model in the "01-speech-commands-mfcc-extraction.ipynb" file.
Below is an example code if you have 9 classes.


#For single word classifier
#wake_word_index = all_targets.index(wake_word)
#y_train = np.equal(y_train, wake_word_index).astype('float64')
#y_val = np.equal(y_val, wake_word_index).astype('float64')
#y_test = np.equal(y_test, wake_word_index).astype('float64')

# For multi word classifier
num_classes = 9  #Change here
y_train = keras.utils.to_categorical(y_train, num_classes)
y_val = keras.utils.to_categorical(y_val, num_classes)
y_test = keras.utils.to_categorical(y_test, num_classes)


3. In the same file, look for the step below. 
Replace the code below. 
Change the number of classes to the number of words you trained the model in the "01-speech-commands-mfcc-extraction.ipynb" file.
Change the activation function to softmax.
Below is an example code if you have 9 classes.

# Build model
# Based on: https://www.geeksforgeeks.org/python-image-classification-using-keras/
model = models.Sequential()
model.add(layers.Conv2D(32, 
                        (2, 2), 
                        activation='relu',
                        input_shape=sample_shape))
model.add(layers.MaxPooling2D(pool_size=(2, 2)))

model.add(layers.Conv2D(32, (2, 2), activation='relu'))
model.add(layers.MaxPooling2D(pool_size=(2, 2)))

model.add(layers.Conv2D(64, (2, 2), activation='relu'))
model.add(layers.MaxPooling2D(pool_size=(2, 2)))

# Classifier
model.add(layers.Flatten())
model.add(layers.Dense(64, activation='relu'))
model.add(layers.Dropout(0.5))

#Change the number to reflect the number of classes.
#Change the activation function to fit single or multiclass.
#Single = 'sigmoid'
#Multiclass = 'softmax'

model.add(layers.Dense(9, activation='softmax'))

**Comment by ShawnHymel** [comment]
Thank you for the example, @mopbop!

@hemsinghBanoth - I have created another demo that relies on Edge Impulse to do the training (if you're open to that tool). It uses MFCCs as features, so it seems to create a much more robust model that what I have in this repo. You can follow the instructions on this repo to run a curation script that allows you to add your own custom keywords: [https://github.com/ShawnHymel/ei-keyword-spotting](https://github.com/ShawnHymel/ei-keyword-spotting).
